AutoNumber/BlobOptimisticDataStore.cs

```diff
@@ -42,34 +49,65 @@ public async Task<string> GetDataAsync(string blockName)
 
             using (var stream = new MemoryStream())
             {
-                await blobReference.DownloadToStreamAsync(stream).ConfigureAwait(false);
+                await blobReference.DownloadToAsync(stream).ConfigureAwait(false);
                 return Encoding.UTF8.GetString(stream.ToArray());
             }
         }
 
-        public async Task<bool> Init()
+        public async Task<bool> InitAsync()
+        {
+            var result = await blobContainer.CreateIfNotExistsAsync().ConfigureAwait(false);
+            return result == null || result.Value != null;

```

Comment:
This looks weird a bit (and it probably is) problem is that the underlying library returns null if the container already exists the second time, see [here ](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs#L1077-L1081)

Reply:
Additional context [here](https://github.com/Azure/azure-sdk-for-net/issues/9758) (I wanted to cry when I read the reason why they're not fixing this 😅 )